### PR TITLE
Finalize DatabaseRotation improvements

### PR DESCRIPTION
**Pull request by KyleAure**
- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

Fail test projects that do not correctly declare they are enrolled in DatabaseRotation

### CODE REVIEW

**Comment by KyleAure** [comment]
#build - personal build

**Comment by KyleAure** [comment]
#build - redo personal build

**Review by gjwatts — DISMISSED** [review]

File: dev/fattest.simplicity/src/componenttest/topology/database/container/DatabaseContainerFactory.java [hunk 1]
```diff
@@ -95,9 +95,8 @@ public static JdbcDatabaseContainer<?> create(DatabaseContainerType defaultType)
         Log.info(c, "create", "System property: fat.bucket.db.type is " + dbProperty);
 
         if (!validateDatabaseRotationFeature) {
-            // TODO throw this exception once WL is updated
-//            throw new IllegalArgumentException("To use a generic database, the FAT must be opted into database rotation by setting 'testedFeatures: " //
-//                                               + databaseRotationTestFeature + "' in the FAT project's bnd.bnd file");
+            throw new IllegalArgumentException("To use a generic database, the FAT must be opted into database rotation by setting 'testedFeatures: " //
```
Comment (KyleAure) [inline dev/fattest.simplicity/src/componenttest/topology/database/container/DatabaseContainerFactory.java]:
```suggestion
            throw new IllegalArgumentException("To use a generic database, the FAT must be opted into database rotation by setting 'tested.features: " //
```

**Comment by KyleAure** [comment]
#build - personal build

**Review by gjwatts — APPROVED** [review]

**Comment by KyleAure** [comment]
RTC Build: 1 known defect, 1 connection issue
Pipeline: Passed

**Pull request merged**